manager/scheduler/constraint.go

```diff
@@ -44,26 +44,36 @@ func (f *ConstraintFilter) Check(n *NodeInfo) bool {
 			if !constraint.Match(n.Description.Hostname) {
 				return false
 			}
+		case "node.role":
```

Comment:
A mild problem though is we don't re-kick the scheduler when labels/roles change.

If you `node.role!=manager` then promote, the tasks are going to remain on the wrong box.

Is this a problem?

/cc @diogomonica @aaronlehmann